src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1771,13 +1772,31 @@ internal void OnPostedOperationError(object sender, Exception e)
                         string[] names = ((ResultListValue)argList).FindAllStrings("name");
                         foreach (var n in names)
                         {
-                            args.Add(new SimpleVariableInformation(n, /*isParam*/ true, null, null));
+                            // If the types of the arguments are requested, get that from a call to -var-create
+                            if (types)
```

Comment:
Doesn't this do *both* costly things now if I specify values && types? If we asked for values, don't we already have the type data here?

Reply:
@chuckries @gregg-miskelly  No, because you would not be in this block if you had both. Line 1762 would be true as if you went through the "simple values" your argList would be a ValueTypeList and it would run through that if block and not this else block.

Reply:
If it helps, without the value, its a NamedResultValue so it falls into the else block.